parser/src/test/java/com/facebook/coresql/parser/sqllogictest/java/SqlLogicTest.java

```diff
@@ -0,0 +1,336 @@
+                int endOfCurrStatement = min(startingIdxOfNextStatement, getIndexOfNextDelimiter(currIdx, linesFromFile));
+                statementsAsListsOfLines.add(linesFromFile.subList(currIdx, endOfCurrStatement));
+                currIdx = startingIdxOfNextStatement;
+            }
+            else {
+                ++currIdx;
+            }
+        }
+        return statementsAsListsOfLines;
+    }
+
+    private SqlLogicTestStatement buildSqlLogicTestStatementFromLines(List<String> statementAsListsOfLines)
+    {
+        SqlLogicTestStatement.Builder builder = SqlLogicTestStatement.builder();
+        StringBuilder statement = new StringBuilder();
+        for (String line : statementAsListsOfLines) {
+            if (isConditionalRecordLine(line)) {
+                processConditionalRecordLine(line, builder);
+            }
+            else if (!isLineBeforeStatement(line)) {
+                statement.append(line);
+            }
+        }
+        String cleanStatement = getCleanedUpStatement(statement);
+        return builder.setStatement(cleanStatement).setError(checkStatementForParserError(cleanStatement)).build();
+    }
+
+    private List<SqlLogicTestStatement> parseSqlLogicTestStatementsFromFile(Path file)
+            throws IOException
+    {
+        List<String> lines = readAllLines(file);
+        List<List<String>> statementsAsListsOfLines = getListOfStatementsAsLines(lines);
+        return statementsAsListsOfLines.stream()
+                .map(this::buildSqlLogicTestStatementFromLines)
+                .filter(this::statementIsAcceptableToCurrDialect)
+                .collect(toList());
+    }
+
+    private List<SqlLogicTestStatement> getErroneousStatements(List<SqlLogicTestStatement> testStatements)
+    {
+        return testStatements.stream()
+                .filter(statement -> checkStatementForParserError(statement.getStatement()).isPresent())
+                .collect(toList());
+    }
+
+    private class SqlLogicTestFileVisitor
+            implements FileVisitor<Path>
+    {
+        @Override
+        public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs)
+        {
+            return CONTINUE;
+        }
+
+        @Override
+        public FileVisitResult visitFile(Path file, BasicFileAttributes attrs)
+                throws IOException
+        {
+            if (isHidden(file)) {
+                return CONTINUE;
+            }
+            if (result.getNumFilesVisited() >= maxNumFilesToVisit) {
+                return TERMINATE;
+            }
+            result.incrementNumFilesVisited();
+            List<SqlLogicTestStatement> testStatements = parseSqlLogicTestStatementsFromFile(file);
+            result.incrementNumStatementsFound(testStatements.size());
+            List<SqlLogicTestStatement> erroneousStatements = getErroneousStatements(testStatements);
+            erroneousStatements.forEach(result::addErroneousStatement);
```

Comment:
If we are going to add the erroneous statements as a list then maybe have addErroneousStatement take in a list?

Reply:
Done.